### PR TITLE
Add a script to add versions to sync configuration

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

Add a new python script that can be used to add a new version to
the configuration of a component.
This script can be used by various projects as part of their
release process to keep the version on the website up-to-date.

Signed-off-by: Andrea Frittoli <andrea.frittoli@gmail.com>
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

/cc @popcor255 
/cc @AlanGreene

### CODE REVIEW

**Comment by popcor255** [comment]
This looks amazing. I wish I setup ci/cd for the tests. You have to give me a sec to test it on my dev environment. lol

File: sync/versions.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+from absl import app
+from absl import flags
+
+import sync
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    'project', 'pipelines', 'Name of the component', short_name='p')
```
Comment (AlanGreene) [inline sync/versions.py:27]:
```suggestion
    'project', 'pipeline', 'Name of the component', short_name='p')
```

File: sync/README.md [hunk 2]
```diff
@@ -94,4 +101,37 @@ To build the docker file
...
+## Usage
+
+```bash
+USAGE: ./sync/versions.py [flags]
+
+flags:
+
+./sync/versions.py:
+  -p,--project: Name of the component
+    (default: 'pipelines')
```
Comment (AlanGreene) [inline sync/README.md:132]:
```suggestion
    (default: 'pipeline')
```

**Review by AlanGreene — CHANGES_REQUESTED** [review]
Useful script! 👍 

Small issue with the default value for `project`, it should be `pipeline` instead of `pipelines`. Should we rename the config file to match / avoid confusion?

It would also be nice it the script output an error on an invalid project name instead of failing silently but that's not a blocker for this PR.

File: sync/README.md (hunk 2 above)
Comment (afrittoli) [reply sync/README.md:132]: Nice catch, thank you!

**Comment by afrittoli** [comment]
> Useful script! 👍
> 
> Small issue with the default value for `project`, it should be `pipeline` instead of `pipelines`. Should we rename the config file to match / avoid confusion?
> 
> It would also be nice if the script output an error on an invalid project name instead of failing silently but that's not a blocker for this PR.

I'll update the script to error in case of project not found

File: sync/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 wget==3.2
-PyYAML==5.1.2
+ruamel.yaml==0.16.12
```
Comment (afrittoli) [inline sync/requirements.txt:2]: This change is so that we can preserve comments in the YAML config when updating them via a python script.

**Comment by afrittoli** [comment]
The script will now log an error and exit with code 1 in case the project cannot be found.

**Review by AlanGreene — APPROVED** [review]
Looks good :shipit:

**Comment by popcor255** [comment]
image
This might be a bug with netlify, but there a duplicates of fields.

**Comment by afrittoli** [comment]
> image
> This might be a bug with netlify, but there a duplicates of fields.

I wonder if this is because I changed the component names in the config, and old files were left around? I though netlify would use fresh envs everytime, but maybe they are not.

It looks good on my laptop, with `letlify dev`- so I don't know which to trust now?

**Comment by AlanGreene** [comment]
Good catch @popcor255, seeing it on the PR deploy preview too:
https://deploy-preview-183--tekton.netlify.app/docs/

Case sensitivity issue perhaps? e.g. `CLI` vs `cli`
Checking the build output locally again

**Comment by AlanGreene** [comment]
See https://github.com/tektoncd/website/tree/81ca55da2a3ff30fcb9ef2abc36eaaa6ac1aceca/content/en/docs
image

With the updated config, content is synced to folders named in lower case and we end up with:
```
Catalog/
Concepts/
Dashboard/
Getting Started/
Pipelines/
_index.md
cli/
getting-started-overview.md
pipeline/
triggers/
```

On a case-sensitive filesystem we would have duplicates for CLI and Triggers too.

**Comment by afrittoli** [comment]
OK - I think the easiest will be to revert the names there then.
The reason for changing the name was to avoid have a name matching that of the project.
When adding a new release I don't want to use the name that is displayed on the website, I'd like to use the git project name instead.
I can change the match to be on `repository` instead of `component`.

**Comment by popcor255** [comment]
I have had this issue as well. Sync does not delete the old docs folder. So, even if we change the env. Everytime we change the sync we would be adding files without removing the older ones. We probably should delete the content folder whenever we redeploy.

**Comment by afrittoli** [comment]
Preview looks good now.

**Pull request merged**